src/Microsoft.AspNet.OData.Shared/DeltaSetOfT.cs

```diff
@@ -266,15 +217,13 @@ internal DeltaSet<TStructuralType> CopyChangedValues(IODataAPIHandler apiHandler
                         {
                             operation = DataModificationOperationKind.Update;
 
-                            if (hasODataId)
-                            {
-                                ODataAPIResponseStatus linkResponseStatus = apiHandlerOfT.TryAddRelatedObject(original, out errorMessage);
+                            ODataAPIResponseStatus linkResponseStatus = apiHandlerOfT.TryAddRelatedObject(original, out errorMessage);
 
-                                if (linkResponseStatus == ODataAPIResponseStatus.Failure)
-                                {
-                                    IDeltaSetItem changedObject = HandleFailedOperation(changedObj, operation, original, errorMessage);
-                                    deltaSet.Add(changedObject);
-                                }
+                            if (linkResponseStatus == ODataAPIResponseStatus.Failure)
```

Comment:
how do you know that the exception was thrown because this method was called twice or the linked object already exists.

Reply:
I think we should throw the exception here and let the customer handle the issue of calling the method twice.